### PR TITLE
forgejo-runner: 3.5.1 -> 4.0.1

### PR DESCRIPTION
**Pull request by emilylange**
https://code.forgejo.org/forgejo/runner/src/tag/v4.0.1/RELEASE-NOTES.md

diff: https://code.forgejo.org/forgejo/runner/compare/v3.5.1...v4.0.1

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alexandru0-dev** [comment]
4.0.1 includes breaking changes.
Do not merge this PR when breaking changes are not allowed, see https://github.com/NixOS/nixpkgs/issues/339153.

(Breaking changes are not allowed until November 8, sry if you're going to wait for this PR to be merged)

**Comment by emilylange** [comment]
> 4.0.1 includes breaking changes. Do not merge this PR when breaking changes are not allowed, see #339153.
> 
> (Breaking changes are not allowed until November 8, sry if you're going to wait for this PR to be merged)

Not sure where you are getting November 8 from, but this is not the correct date.
On November 8 the only branch that will be unrestricted from the breaking change restriction is `staging`.
This PR here is targeting `master`, not `staging`.
`master` will be unrestricted after branch-off, on November 14.

Additionally, the breaking change you are referring to is technically a security fix.
I just did not feel like flagging it as such via the `1. severity: security` label.

And finally, `forgejo-runner` is a leaf package. No other package depends on it.

I am not willing to go into `24.11` with an EOL version.

---

I'll look into why the VM test is all of a sudden failing and try to fix it.
I can't remember it failing when I created this PR (otherwise I wouldn't have ticked the "Tested, as applicable" checkbox), but I am able to reproduce it consistently on the machine I am currently on.

When this is sorted, I'll go forward with merging this PR.
If for reasons that aren't clear to me yet this is deemed the wrong call here, I am fine with a revert and will look into the alternatives.

**Review by adamcstephens — APPROVED** [review]

**Comment by emilylange** [comment]
Test got fixed in #353670

@ofborg test forgejo-runner

**Comment by emilylange** [comment]
@ofborg build forgejo-runner.tests

**Pull request merged**
